<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<title></title>

<script type="text/javascript">
window.onload = function() {
  var imgs = document.getElementsByTagName('img'), i, img;
  for (i = 0; i < imgs.length; i++) {
    img = imgs[i];
    // center an image if it is the only element of its parent
    if (img.parentElement.childElementCount === 1)
      img.parentElement.style.textAlign = 'center';
  }
};
</script>

<!-- Styles for R syntax highlighter -->
<style type="text/css">
   pre .operator,
   pre .paren {
     color: rgb(104, 118, 135)
   }

   pre .literal {
     color: #990073
   }

   pre .number {
     color: #099;
   }

   pre .comment {
     color: #998;
     font-style: italic
   }

   pre .keyword {
     color: #900;
     font-weight: bold
   }

   pre .identifier {
     color: rgb(0, 0, 0);
   }

   pre .string {
     color: #d14;
   }
</style>

<!-- R syntax highlighter -->
<script type="text/javascript">
var hljs=new function(){function m(p){return p.replace(/&/gm,"&amp;").replace(/</gm,"&lt;")}function f(r,q,p){return RegExp(q,"m"+(r.cI?"i":"")+(p?"g":""))}function b(r){for(var p=0;p<r.childNodes.length;p++){var q=r.childNodes[p];if(q.nodeName=="CODE"){return q}if(!(q.nodeType==3&&q.nodeValue.match(/\s+/))){break}}}function h(t,s){var p="";for(var r=0;r<t.childNodes.length;r++){if(t.childNodes[r].nodeType==3){var q=t.childNodes[r].nodeValue;if(s){q=q.replace(/\n/g,"")}p+=q}else{if(t.childNodes[r].nodeName=="BR"){p+="\n"}else{p+=h(t.childNodes[r])}}}if(/MSIE [678]/.test(navigator.userAgent)){p=p.replace(/\r/g,"\n")}return p}function a(s){var r=s.className.split(/\s+/);r=r.concat(s.parentNode.className.split(/\s+/));for(var q=0;q<r.length;q++){var p=r[q].replace(/^language-/,"");if(e[p]){return p}}}function c(q){var p=[];(function(s,t){for(var r=0;r<s.childNodes.length;r++){if(s.childNodes[r].nodeType==3){t+=s.childNodes[r].nodeValue.length}else{if(s.childNodes[r].nodeName=="BR"){t+=1}else{if(s.childNodes[r].nodeType==1){p.push({event:"start",offset:t,node:s.childNodes[r]});t=arguments.callee(s.childNodes[r],t);p.push({event:"stop",offset:t,node:s.childNodes[r]})}}}}return t})(q,0);return p}function k(y,w,x){var q=0;var z="";var s=[];function u(){if(y.length&&w.length){if(y[0].offset!=w[0].offset){return(y[0].offset<w[0].offset)?y:w}else{return w[0].event=="start"?y:w}}else{return y.length?y:w}}function t(D){var A="<"+D.nodeName.toLowerCase();for(var B=0;B<D.attributes.length;B++){var C=D.attributes[B];A+=" "+C.nodeName.toLowerCase();if(C.value!==undefined&&C.value!==false&&C.value!==null){A+='="'+m(C.value)+'"'}}return A+">"}while(y.length||w.length){var v=u().splice(0,1)[0];z+=m(x.substr(q,v.offset-q));q=v.offset;if(v.event=="start"){z+=t(v.node);s.push(v.node)}else{if(v.event=="stop"){var p,r=s.length;do{r--;p=s[r];z+=("</"+p.nodeName.toLowerCase()+">")}while(p!=v.node);s.splice(r,1);while(r<s.length){z+=t(s[r]);r++}}}}return z+m(x.substr(q))}function j(){function q(x,y,v){if(x.compiled){return}var u;var s=[];if(x.k){x.lR=f(y,x.l||hljs.IR,true);for(var w in x.k){if(!x.k.hasOwnProperty(w)){continue}if(x.k[w] instanceof Object){u=x.k[w]}else{u=x.k;w="keyword"}for(var r in u){if(!u.hasOwnProperty(r)){continue}x.k[r]=[w,u[r]];s.push(r)}}}if(!v){if(x.bWK){x.b="\\b("+s.join("|")+")\\s"}x.bR=f(y,x.b?x.b:"\\B|\\b");if(!x.e&&!x.eW){x.e="\\B|\\b"}if(x.e){x.eR=f(y,x.e)}}if(x.i){x.iR=f(y,x.i)}if(x.r===undefined){x.r=1}if(!x.c){x.c=[]}x.compiled=true;for(var t=0;t<x.c.length;t++){if(x.c[t]=="self"){x.c[t]=x}q(x.c[t],y,false)}if(x.starts){q(x.starts,y,false)}}for(var p in e){if(!e.hasOwnProperty(p)){continue}q(e[p].dM,e[p],true)}}function d(B,C){if(!j.called){j();j.called=true}function q(r,M){for(var L=0;L<M.c.length;L++){if((M.c[L].bR.exec(r)||[null])[0]==r){return M.c[L]}}}function v(L,r){if(D[L].e&&D[L].eR.test(r)){return 1}if(D[L].eW){var M=v(L-1,r);return M?M+1:0}return 0}function w(r,L){return L.i&&L.iR.test(r)}function K(N,O){var M=[];for(var L=0;L<N.c.length;L++){M.push(N.c[L].b)}var r=D.length-1;do{if(D[r].e){M.push(D[r].e)}r--}while(D[r+1].eW);if(N.i){M.push(N.i)}return f(O,M.join("|"),true)}function p(M,L){var N=D[D.length-1];if(!N.t){N.t=K(N,E)}N.t.lastIndex=L;var r=N.t.exec(M);return r?[M.substr(L,r.index-L),r[0],false]:[M.substr(L),"",true]}function z(N,r){var L=E.cI?r[0].toLowerCase():r[0];var M=N.k[L];if(M&&M instanceof Array){return M}return false}function F(L,P){L=m(L);if(!P.k){return L}var r="";var O=0;P.lR.lastIndex=0;var M=P.lR.exec(L);while(M){r+=L.substr(O,M.index-O);var N=z(P,M);if(N){x+=N[1];r+='<span class="'+N[0]+'">'+M[0]+"</span>"}else{r+=M[0]}O=P.lR.lastIndex;M=P.lR.exec(L)}return r+L.substr(O,L.length-O)}function J(L,M){if(M.sL&&e[M.sL]){var r=d(M.sL,L);x+=r.keyword_count;return r.value}else{return F(L,M)}}function I(M,r){var L=M.cN?'<span class="'+M.cN+'">':"";if(M.rB){y+=L;M.buffer=""}else{if(M.eB){y+=m(r)+L;M.buffer=""}else{y+=L;M.buffer=r}}D.push(M);A+=M.r}function G(N,M,Q){var R=D[D.length-1];if(Q){y+=J(R.buffer+N,R);return false}var P=q(M,R);if(P){y+=J(R.buffer+N,R);I(P,M);return P.rB}var L=v(D.length-1,M);if(L){var O=R.cN?"</span>":"";if(R.rE){y+=J(R.buffer+N,R)+O}else{if(R.eE){y+=J(R.buffer+N,R)+O+m(M)}else{y+=J(R.buffer+N+M,R)+O}}while(L>1){O=D[D.length-2].cN?"</span>":"";y+=O;L--;D.length--}var r=D[D.length-1];D.length--;D[D.length-1].buffer="";if(r.starts){I(r.starts,"")}return R.rE}if(w(M,R)){throw"Illegal"}}var E=e[B];var D=[E.dM];var A=0;var x=0;var y="";try{var s,u=0;E.dM.buffer="";do{s=p(C,u);var t=G(s[0],s[1],s[2]);u+=s[0].length;if(!t){u+=s[1].length}}while(!s[2]);if(D.length>1){throw"Illegal"}return{r:A,keyword_count:x,value:y}}catch(H){if(H=="Illegal"){return{r:0,keyword_count:0,value:m(C)}}else{throw H}}}function g(t){var p={keyword_count:0,r:0,value:m(t)};var r=p;for(var q in e){if(!e.hasOwnProperty(q)){continue}var s=d(q,t);s.language=q;if(s.keyword_count+s.r>r.keyword_count+r.r){r=s}if(s.keyword_count+s.r>p.keyword_count+p.r){r=p;p=s}}if(r.language){p.second_best=r}return p}function i(r,q,p){if(q){r=r.replace(/^((<[^>]+>|\t)+)/gm,function(t,w,v,u){return w.replace(/\t/g,q)})}if(p){r=r.replace(/\n/g,"<br>")}return r}function n(t,w,r){var x=h(t,r);var v=a(t);var y,s;if(v){y=d(v,x)}else{return}var q=c(t);if(q.length){s=document.createElement("pre");s.innerHTML=y.value;y.value=k(q,c(s),x)}y.value=i(y.value,w,r);var u=t.className;if(!u.match("(\\s|^)(language-)?"+v+"(\\s|$)")){u=u?(u+" "+v):v}if(/MSIE [678]/.test(navigator.userAgent)&&t.tagName=="CODE"&&t.parentNode.tagName=="PRE"){s=t.parentNode;var p=document.createElement("div");p.innerHTML="<pre><code>"+y.value+"</code></pre>";t=p.firstChild.firstChild;p.firstChild.cN=s.cN;s.parentNode.replaceChild(p.firstChild,s)}else{t.innerHTML=y.value}t.className=u;t.result={language:v,kw:y.keyword_count,re:y.r};if(y.second_best){t.second_best={language:y.second_best.language,kw:y.second_best.keyword_count,re:y.second_best.r}}}function o(){if(o.called){return}o.called=true;var r=document.getElementsByTagName("pre");for(var p=0;p<r.length;p++){var q=b(r[p]);if(q){n(q,hljs.tabReplace)}}}function l(){if(window.addEventListener){window.addEventListener("DOMContentLoaded",o,false);window.addEventListener("load",o,false)}else{if(window.attachEvent){window.attachEvent("onload",o)}else{window.onload=o}}}var e={};this.LANGUAGES=e;this.highlight=d;this.highlightAuto=g;this.fixMarkup=i;this.highlightBlock=n;this.initHighlighting=o;this.initHighlightingOnLoad=l;this.IR="[a-zA-Z][a-zA-Z0-9_]*";this.UIR="[a-zA-Z_][a-zA-Z0-9_]*";this.NR="\\b\\d+(\\.\\d+)?";this.CNR="\\b(0[xX][a-fA-F0-9]+|(\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)";this.BNR="\\b(0b[01]+)";this.RSR="!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|\\.|-|-=|/|/=|:|;|<|<<|<<=|<=|=|==|===|>|>=|>>|>>=|>>>|>>>=|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~";this.ER="(?![\\s\\S])";this.BE={b:"\\\\.",r:0};this.ASM={cN:"string",b:"'",e:"'",i:"\\n",c:[this.BE],r:0};this.QSM={cN:"string",b:'"',e:'"',i:"\\n",c:[this.BE],r:0};this.CLCM={cN:"comment",b:"//",e:"$"};this.CBLCLM={cN:"comment",b:"/\\*",e:"\\*/"};this.HCM={cN:"comment",b:"#",e:"$"};this.NM={cN:"number",b:this.NR,r:0};this.CNM={cN:"number",b:this.CNR,r:0};this.BNM={cN:"number",b:this.BNR,r:0};this.inherit=function(r,s){var p={};for(var q in r){p[q]=r[q]}if(s){for(var q in s){p[q]=s[q]}}return p}}();hljs.LANGUAGES.cpp=function(){var a={keyword:{"false":1,"int":1,"float":1,"while":1,"private":1,"char":1,"catch":1,"export":1,virtual:1,operator:2,sizeof:2,dynamic_cast:2,typedef:2,const_cast:2,"const":1,struct:1,"for":1,static_cast:2,union:1,namespace:1,unsigned:1,"long":1,"throw":1,"volatile":2,"static":1,"protected":1,bool:1,template:1,mutable:1,"if":1,"public":1,friend:2,"do":1,"return":1,"goto":1,auto:1,"void":2,"enum":1,"else":1,"break":1,"new":1,extern:1,using:1,"true":1,"class":1,asm:1,"case":1,typeid:1,"short":1,reinterpret_cast:2,"default":1,"double":1,register:1,explicit:1,signed:1,typename:1,"try":1,"this":1,"switch":1,"continue":1,wchar_t:1,inline:1,"delete":1,alignof:1,char16_t:1,char32_t:1,constexpr:1,decltype:1,noexcept:1,nullptr:1,static_assert:1,thread_local:1,restrict:1,_Bool:1,complex:1},built_in:{std:1,string:1,cin:1,cout:1,cerr:1,clog:1,stringstream:1,istringstream:1,ostringstream:1,auto_ptr:1,deque:1,list:1,queue:1,stack:1,vector:1,map:1,set:1,bitset:1,multiset:1,multimap:1,unordered_set:1,unordered_map:1,unordered_multiset:1,unordered_multimap:1,array:1,shared_ptr:1}};return{dM:{k:a,i:"</",c:[hljs.CLCM,hljs.CBLCLM,hljs.QSM,{cN:"string",b:"'\\\\?.",e:"'",i:"."},{cN:"number",b:"\\b(\\d+(\\.\\d*)?|\\.\\d+)(u|U|l|L|ul|UL|f|F)"},hljs.CNM,{cN:"preprocessor",b:"#",e:"$"},{cN:"stl_container",b:"\\b(deque|list|queue|stack|vector|map|set|bitset|multiset|multimap|unordered_map|unordered_set|unordered_multiset|unordered_multimap|array)\\s*<",e:">",k:a,r:10,c:["self"]}]}}}();hljs.LANGUAGES.r={dM:{c:[hljs.HCM,{cN:"number",b:"\\b0[xX][0-9a-fA-F]+[Li]?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+(?:[eE][+\\-]?\\d*)?L\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+\\.(?!\\d)(?:i\\b)?",e:hljs.IMMEDIATE_RE,r:1},{cN:"number",b:"\\b\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"keyword",b:"(?:tryCatch|library|setGeneric|setGroupGeneric)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\.",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\d+(?![\\w.])",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\b(?:function)",e:hljs.IMMEDIATE_RE,r:2},{cN:"keyword",b:"(?:if|in|break|next|repeat|else|for|return|switch|while|try|stop|warning|require|attach|detach|source|setMethod|setClass)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"literal",b:"(?:NA|NA_integer_|NA_real_|NA_character_|NA_complex_)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"literal",b:"(?:NULL|TRUE|FALSE|T|F|Inf|NaN)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"identifier",b:"[a-zA-Z.][a-zA-Z0-9._]*\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"<\\-(?!\\s*\\d)",e:hljs.IMMEDIATE_RE,r:2},{cN:"operator",b:"\\->|<\\-",e:hljs.IMMEDIATE_RE,r:1},{cN:"operator",b:"%%|~",e:hljs.IMMEDIATE_RE},{cN:"operator",b:">=|<=|==|!=|\\|\\||&&|=|\\+|\\-|\\*|/|\\^|>|<|!|&|\\||\\$|:",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"%",e:"%",i:"\\n",r:1},{cN:"identifier",b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[hljs.BE],r:0},{cN:"string",b:"'",e:"'",c:[hljs.BE],r:0},{cN:"paren",b:"[[({\\])}]",e:hljs.IMMEDIATE_RE,r:0}]}};
hljs.initHighlightingOnLoad();
</script>



<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 13px;
}

body {
  max-width: 800px;
  margin: auto;
  padding: 1em;
  line-height: 20px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 {
   font-size:2.2em;
}

h2 {
   font-size:1.8em;
}

h3 {
   font-size:1.4em;
}

h4 {
   font-size:1.0em;
}

h5 {
   font-size:0.9em;
}

h6 {
   font-size:0.8em;
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre, img {
  max-width: 100%;
}
pre {
  overflow-x: auto;
}
pre code {
   display: block; padding: 0.5em;
}

code {
  font-size: 92%;
  border: 1px solid #ccc;
}

code[class] {
  background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * {
      background: transparent !important;
      color: black !important;
      filter:none !important;
      -ms-filter: none !important;
   }

   body {
      font-size:12pt;
      max-width:100%;
   }

   a, a:visited {
      text-decoration: underline;
   }

   hr {
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote {
      padding-right: 1em;
      page-break-inside: avoid;
   }

   tr, img {
      page-break-inside: avoid;
   }

   img {
      max-width: 100% !important;
   }

   @page :left {
      margin: 15mm 20mm 15mm 10mm;
   }

   @page :right {
      margin: 15mm 10mm 15mm 20mm;
   }

   p, h2, h3 {
      orphans: 3; widows: 3;
   }

   h2, h3 {
      page-break-after: avoid;
   }
}
</style>



</head>

<body>
<pre><code class="r">library(&quot;here&quot;)

source(here(&quot;code&quot;, &quot;models&quot;, &quot;run_gsea.R&quot;), echo = TRUE)
</code></pre>

<pre><code>## 
## &gt; ## ---- setup_env
## &gt; library(&quot;here&quot;)
## 
## &gt; source(here(&quot;code&quot;, &quot;data&quot;, &quot;create_ubiopred_eset.R&quot;), echo = TRUE)
## 
## &gt; ## ---- setup_env
## &gt; options(rgl.useNULL = TRUE) # prevent X11 display warning
## 
## &gt; library(&quot;iCheck&quot;)
</code></pre>

<pre><code>## Loading required package: Biobase
</code></pre>

<pre><code>## Loading required package: BiocGenerics
</code></pre>

<pre><code>## Loading required package: parallel
</code></pre>

<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;
</code></pre>

<pre><code>## The following objects are masked from &#39;package:parallel&#39;:
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB
</code></pre>

<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs
</code></pre>

<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, cbind, colMeans,
##     colnames, colSums, do.call, duplicated, eval, evalq, Filter,
##     Find, get, grep, grepl, intersect, is.unsorted, lapply,
##     lengths, Map, mapply, match, mget, order, paste, pmax,
##     pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce,
##     rowMeans, rownames, rowSums, sapply, setdiff, sort, table,
##     tapply, union, unique, unsplit, which, which.max, which.min
</code></pre>

<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.
</code></pre>

<pre><code>## Loading required package: lumi
</code></pre>

<pre><code>## Setting options(&#39;download.file.method.GEOquery&#39;=&#39;auto&#39;)
</code></pre>

<pre><code>## Setting options(&#39;GEOquery.inmemory.gpl&#39;=FALSE)
</code></pre>

<pre><code>## Loading required package: gplots
</code></pre>

<pre><code>## 
## Attaching package: &#39;gplots&#39;
</code></pre>

<pre><code>## The following object is masked from &#39;package:stats&#39;:
## 
##     lowess
</code></pre>

<pre><code>## 
## &gt; library(&quot;CellMix&quot;)
</code></pre>

<pre><code>## Loading required package: pkgmaker
</code></pre>

<pre><code>## Loading required package: registry
</code></pre>

<pre><code>## 
## Attaching package: &#39;pkgmaker&#39;
</code></pre>

<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     isNamespaceLoaded
</code></pre>

<pre><code>## Loading required package: NMF
</code></pre>

<pre><code>## Loading required package: rngtools
</code></pre>

<pre><code>## Loading required package: cluster
</code></pre>

<pre><code>## NMF - BioConductor layer [OK] | Shared memory capabilities [NO: synchronicity] | Cores 2/3
</code></pre>

<pre><code>##   To enable shared memory capabilities, try: install.extras(&#39;
## NMF
## &#39;)
</code></pre>

<pre><code>## Loading required package: csSAM
</code></pre>

<pre><code>## Loading required package: compiler
</code></pre>

<pre><code>## Loading required package: stringr
</code></pre>

<pre><code>## Loading required package: GSEABase
</code></pre>

<pre><code>## Loading required package: annotate
</code></pre>

<pre><code>## Loading required package: AnnotationDbi
</code></pre>

<pre><code>## Loading required package: stats4
</code></pre>

<pre><code>## Loading required package: IRanges
</code></pre>

<pre><code>## Loading required package: S4Vectors
</code></pre>

<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;
</code></pre>

<pre><code>## The following object is masked from &#39;package:NMF&#39;:
## 
##     nrun
</code></pre>

<pre><code>## The following object is masked from &#39;package:pkgmaker&#39;:
## 
##     new2
</code></pre>

<pre><code>## The following object is masked from &#39;package:gplots&#39;:
## 
##     space
</code></pre>

<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     expand.grid
</code></pre>

<pre><code>## Loading required package: XML
</code></pre>

<pre><code>## Loading required package: graph
</code></pre>

<pre><code>## 
## Attaching package: &#39;graph&#39;
</code></pre>

<pre><code>## The following object is masked from &#39;package:XML&#39;:
## 
##     addNode
</code></pre>

<pre><code>## The following object is masked from &#39;package:stringr&#39;:
## 
##     boundary
</code></pre>

<pre><code>## Warning: replacing previous import &#39;graph::boundary&#39; by &#39;stringr::boundary&#39;
## when loading &#39;CellMix&#39;
</code></pre>

<pre><code>## 
## Attaching package: &#39;CellMix&#39;
</code></pre>

<pre><code>## The following object is masked from &#39;package:IRanges&#39;:
## 
##     reverse
</code></pre>

<pre><code>## 
## &gt; library(&quot;dplyr&quot;)
</code></pre>

<pre><code>## 
## Attaching package: &#39;dplyr&#39;
</code></pre>

<pre><code>## The following objects are masked from &#39;package:CellMix&#39;:
## 
##     combine, intersect
</code></pre>

<pre><code>## The following objects are masked from &#39;package:GSEABase&#39;:
## 
##     intersect, setdiff, union
</code></pre>

<pre><code>## The following object is masked from &#39;package:graph&#39;:
## 
##     union
</code></pre>

<pre><code>## The following object is masked from &#39;package:AnnotationDbi&#39;:
## 
##     select
</code></pre>

<pre><code>## The following objects are masked from &#39;package:IRanges&#39;:
## 
##     collapse, desc, intersect, setdiff, slice, union
</code></pre>

<pre><code>## The following objects are masked from &#39;package:S4Vectors&#39;:
## 
##     first, intersect, rename, setdiff, setequal, union
</code></pre>

<pre><code>## The following object is masked from &#39;package:lumi&#39;:
## 
##     combine
</code></pre>

<pre><code>## The following object is masked from &#39;package:Biobase&#39;:
## 
##     combine
</code></pre>

<pre><code>## The following objects are masked from &#39;package:BiocGenerics&#39;:
## 
##     combine, intersect, setdiff, union
</code></pre>

<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag
</code></pre>

<pre><code>## The following object is masked from &#39;.env&#39;:
## 
##     n
</code></pre>

<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union
</code></pre>

<pre><code>## 
## &gt; library(&quot;stringr&quot;)
## 
## &gt; library(&quot;broom&quot;)
## 
## &gt; library(&quot;forcats&quot;)
## 
## &gt; library(&quot;magrittr&quot;)
## 
## &gt; library(&quot;genefilter&quot;)
## 
## &gt; library(&quot;devtools&quot;)
## 
## &gt; # custom annotation package:
## &gt; # see &quot;make_hthgu133pluspm_db.R&quot;
## &gt; library(&quot;hthgu133pluspm.db&quot;)
</code></pre>

<pre><code>## Loading required package: org.Hs.eg.db
</code></pre>

<pre><code>## 
</code></pre>

<pre><code>## 
</code></pre>

<pre><code>## 
## &gt; ## ---- load_data
## &gt; base_data_dir &lt;- &quot;data/external&quot;
## 
## &gt; load(file.path(base_data_dir, &quot;GSE69683/data/GSE69683.rda&quot;))
## 
## &gt; es_ubiopred_wb &lt;- GSE69683[[1]]
## 
## &gt; # annotation(es_ubiopred_wb) &lt;- &quot;GPL13158&quot; # aka Affymetrix HT HG-U133+ PM Array
## &gt; annotation(es_ubiopred_wb) &lt;- &quot;hthgu133pluspm&quot;
## 
## &gt; rm(list = ls() %&gt;% extract(!grepl(&quot;^es_&quot;, .))) # clean up
## 
## &gt; ## ---- setup_pheno_data
## &gt; pData(es_ubiopred_wb)$cohort &lt;- es_ubiopred_wb %&gt;% pData %&gt;% use_series(characteristics_ch1) %&gt;% as.character %&gt;% str_rep .... [TRUNCATED] 
## 
## &gt; pData(es_ubiopred_wb)$sex &lt;- es_ubiopred_wb %&gt;% pData %&gt;% use_series(characteristics_ch1.1) %&gt;% as.character %&gt;% str_replace(&quot;gender: &quot;, &quot;&quot;)
## 
## &gt; pData(es_ubiopred_wb)$race &lt;- es_ubiopred_wb %&gt;% pData %&gt;% use_series(characteristics_ch1.2) %&gt;% as.character %&gt;% str_replace(&quot;race: &quot;, &quot;&quot;) %&gt;% fact .... [TRUNCATED] 
## 
## &gt; pData(es_ubiopred_wb)$title &lt;- es_ubiopred_wb %&gt;% pData %&gt;% use_series(title) %&gt;% as.character
## 
## &gt; pData(es_ubiopred_wb)$geo_accession &lt;- es_ubiopred_wb %&gt;% pData %&gt;% use_series(geo_accession) %&gt;% as.character
## 
## &gt; pData(es_ubiopred_wb)$asthma &lt;- es_ubiopred_wb %&gt;% pData %&gt;% use_series(cohort) %in% c(&quot;Severe asthma, non-smoking&quot;, &quot;Moderate asthma, non-smoking&quot;, .... [TRUNCATED] 
## 
## &gt; pData(es_ubiopred_wb)$severe &lt;- es_ubiopred_wb %&gt;% pData %&gt;% use_series(cohort) %in% c(&quot;Severe asthma, non-smoking&quot;, &quot;Severe asthma, smoking&quot;)
## 
## &gt; pData(es_ubiopred_wb)$smoker &lt;- es_ubiopred_wb %&gt;% pData %&gt;% use_series(cohort) %in% c(&quot;Severe asthma, smoking&quot;)
## 
## &gt; ## ---- setup_feature_data
## &gt; fData(es_ubiopred_wb)$ID &lt;- fData(es_ubiopred_wb)$ID %&gt;% as.character
## 
## &gt; fData(es_ubiopred_wb)$symbol &lt;- es_ubiopred_wb %&gt;% fData %&gt;% use_series(`Gene Symbol`) %&gt;% as.character # nb: &#39;DEFA1 /// DEFA1B /// DEFA3&#39;
## 
## &gt; fData(es_ubiopred_wb)$symbol2 &lt;- unlist(mget(fData(es_ubiopred_wb)$ID, hthgu133pluspmSYMBOL)) # somewhat diff from original GEO anno data
## 
## &gt; fData(es_ubiopred_wb)$entrez &lt;- unlist(mget(fData(es_ubiopred_wb)$ID, hthgu133pluspmENTREZID))
## 
## &gt; temp_chr_df &lt;- mget(fData(es_ubiopred_wb)$ID, hthgu133pluspmCHR) %&gt;% unlist %&gt;% data_frame(ID = names(.), chromosome = .) %&gt;% as.data.frame
## 
## &gt; fData(es_ubiopred_wb) %&lt;&gt;% left_join(temp_chr_df)
</code></pre>

<pre><code>## Joining, by = &quot;ID&quot;
</code></pre>

<pre><code>## 
## &gt; rownames(fData(es_ubiopred_wb)) &lt;- fData(es_ubiopred_wb)$ID # put rownames back!
## 
## &gt; rm(temp_chr_df)
## 
## &gt; ## ---- make_gx_pcs_wb
## &gt; res_pca_ubiopred_wb &lt;- getPCAFunc(es = es_ubiopred_wb, labelVariable = &quot;title&quot;, requireLog2 = FALSE, corFlag = TRUE)
## 
## &gt; # merge principal components (PC) of gene expression data into phenotype data
## &gt; pDat_wb &lt;- es_ubiopred_wb %&gt;% pData %&gt;% tibble::rownames_to_column()
## 
## &gt; pcDat_wb &lt;- res_pca_ubiopred_wb %&gt;% use_series(pcs) %&gt;% use_series(x) %&gt;% as.data.frame %&gt;%
## +     set_colnames(paste0(&quot;gx&quot;, colnames(.))) %&gt;% tibble .... [TRUNCATED] 
## 
## &gt; pData(es_ubiopred_wb) &lt;- pDat_wb %&gt;% left_join(pcDat_wb, by = c(&quot;title&quot; = &quot;rowname&quot;)) %&gt;%
## +     set_rownames(pDat_wb %&gt;% use_series(rowname)) %&gt;% dp .... [TRUNCATED] 
## 
## &gt; ## ---- estimate_wb_cbcs
## &gt; em1 &lt;- ExpressionMix(es_ubiopred_wb)
## 
## &gt; res1 &lt;- gedBlood(em1, verbose = TRUE, normalize = &quot;none&quot;)
## Loading basis signature from Abbas et al. (2009) ... OK [359 features x 17 cell types]
## Estimating proportions for blood cell subset(s): WB
## Mapping signature ids onto target ids (method: auto) ... OK [218 features x 17 cell types]
## Limit/reorder to common set of features ... OK [218 features x 17 cell types]
## Checking data dimension compatibility ... OK [218 features x 17 cell types]
## Using cell type signatures: &#39;Th&#39;, &#39;Th act&#39;, ..., &#39;neutro&#39; [17 total]
## Checking log-scale ... data:YES - signatures:NO
## Applying log-transform to signatures (base 2) ... OK
## Normalizing signatures and target together (method: none) ... SKIP
##  Using ged algorithm: &quot;lsfit&quot;
##   Estimating cell proportions from cell-specific signatures [lsfit: ls]
##  Timing:
##    user  system elapsed 
##  32.979   0.008  33.014 
##  GED final wrap up ...  OK
## 
## &gt; ecbc1 &lt;- asCBC(res1)
## 
## &gt; cbc_phenos1 &lt;- ecbc1 %&gt;% 
## +     coef %&gt;% 
## +     t %&gt;%
## +     as_data_frame %&gt;% 
## +     mutate(geo_accession = ecbc1 %&gt;% coef %&gt;% colnames,
## +           .... [TRUNCATED] 
## 
## &gt; pData(es_ubiopred_wb) %&lt;&gt;% left_join(cbc_phenos1)
</code></pre>

<pre><code>## Joining, by = &quot;geo_accession&quot;
</code></pre>

<pre><code>## 
## &gt; ## ---- clean_up_env
## &gt; rm(list = ls() %&gt;% extract(grepl(&quot;^es_ubiopred|^res_&quot;, .) %&gt;% not))
## 
## &gt; library(&quot;fgsea&quot;)
</code></pre>

<pre><code>## Loading required package: Rcpp
</code></pre>

<pre><code>## 
## &gt; my.seed &lt;- 40799039 # a fixed random number seed to aid reproducibility
## 
## &gt; set.seed(my.seed)
## 
## &gt; rundate_gsea &lt;- format(Sys.Date(), format = &quot;%Y%m%d&quot;)
## 
## &gt; ## ---- run_dge
## &gt; # cases vs. ctrls. (all subjects)
## &gt; dge1 &lt;- lmFitWrapper(es_ubiopred_wb, 
## +                      formula = ~asthma + severe + smok .... [TRUNCATED] 
## dim(dat)&gt;&gt;
## [1] 54715   498
## 
## Running lmFit...
## Running eBayes...
## Preparing output...
##          probeIDs geneSymbols  chr     stats         pval        p.adj
## 1    209124_PM_at       MYD88    3  6.137541 1.719762e-09 9.409679e-05
## 2  227280_PM_s_at      CCNYL1    2  5.931601 5.639866e-09 1.542926e-04
## 3    232441_PM_at        KRR1   12 -5.603745 3.487982e-08 4.507697e-04
## 4    226999_PM_at       RNPC3    1 -5.579693 3.973425e-08 4.507697e-04
## 5    201705_PM_at       PSMD7   16  5.573025 4.119252e-08 4.507697e-04
## 6    235645_PM_at       ESCO1   18 -5.471018 7.117843e-08 5.830729e-04
## 7   1566342_PM_at                6  5.462199 7.459582e-08 5.830729e-04
## 8    239629_PM_at       CFLAR    2 -5.429584 8.867353e-08 6.064715e-04
## 9    244822_PM_at               21 -5.392511 1.078158e-07 6.554601e-04
## 10   202026_PM_at        SDHD   11  5.319076 1.582718e-07 8.276226e-04
## 11   235216_PM_at       ESCO1   18 -5.309449 1.663867e-07 8.276226e-04
## 12   214814_PM_at      YTHDC1    4 -5.239576 2.386433e-07 1.026810e-03
## 13   219969_PM_at       TXLNG    X -5.227417 2.539965e-07 1.026810e-03
## 14   241320_PM_at             &lt;NA&gt; -5.209546 2.783132e-07 1.026810e-03
## 15   226219_PM_at    ARHGAP30    1  5.207319 2.814978e-07 1.026810e-03
## 16    40189_PM_at         SET    9  5.180730 3.223432e-07 1.039913e-03
## 17   229410_PM_at     SLC35E1   19  5.180268 3.231020e-07 1.039913e-03
## 18   211074_PM_at       FOLR1   11 -5.158299 3.612160e-07 1.051075e-03
## 19   215018_PM_at    KIAA1731   11 -5.145831 3.847447e-07 1.051075e-03
## 20 221705_PM_s_at       SIKE1    1 -5.125883 4.255087e-07 1.051075e-03
##      pos
## 1  18539
## 2  36536
## 3  41696
## 4  36255
## 5  11154
## 6  44895
## 7   8251
## 8  48879
## 9  54073
## 10 11475
## 11 44466
## 12 24110
## 13 29254
## 14 50570
## 15 35476
## 16 54349
## 17 38665
## 18 20439
## 19 24313
## 20 30986
## 
## pvalue quantiles for intercept and covariates&gt;&gt;
##         (Intercept)   asthmaTRUE   severeTRUE   smokerTRUE    sexmale
## min    0.000000e+00 1.719762e-09 7.107525e-10 2.604129e-14 0.00000000
## 25%    2.546901e-97 9.045413e-02 1.240844e-01 1.727951e-01 0.05351697
## median 5.360849e-67 3.275531e-01 3.767533e-01 4.250300e-01 0.27780405
## 75%    5.225016e-47 6.457142e-01 6.825190e-01 7.088523e-01 0.61493780
## max    9.861531e-01 9.999890e-01 9.999738e-01 9.999963e-01 0.99999435
##                gxPC1         gxPC2      ESTNEUT
## min    4.152111e-228 1.386093e-102 2.644747e-81
## 25%     1.116799e-32  4.692088e-13 9.342563e-07
## median  2.742412e-13  2.163824e-05 7.572454e-03
## 75%     2.202629e-04  4.130509e-02 2.173101e-01
## max     9.998273e-01  9.999473e-01 9.997582e-01
## 
## formula&gt;&gt;
## ~asthma + severe + smoker + sex + gxPC1 + gxPC2 + ESTNEUT
## 
## covariate of interest is  asthma 
## Number of tests= 54715 
## Number of arrays= 498 
## Number of significant tests (raw p-value &lt;  0.05 )= 10128 
## Number of significant tests after p-value adjustments= 2305 
## 
## 
## **********************************************
## 
## &gt; # cases vs. ctrls. (no smokers)
## &gt; dge2 &lt;- lmFitWrapper(es_ubiopred_wb[, !es_ubiopred_wb$smoker], 
## +                      formula = ~asthma + severe  .... [TRUNCATED] 
## dim(dat)&gt;&gt;
## [1] 54715   410
## 
## Running lmFit...
## Running eBayes...
## Preparing output...
##           probeIDs geneSymbols  chr     stats         pval        p.adj
## 1     209124_PM_at       MYD88    3  6.167768 1.669291e-09 9.133527e-05
## 2   227280_PM_s_at      CCNYL1    2  5.876302 8.726027e-09 2.387223e-04
## 3     232441_PM_at        KRR1   12 -5.706957 2.215427e-08 3.656172e-04
## 4     235645_PM_at       ESCO1   18 -5.672353 2.672885e-08 3.656172e-04
## 5     226999_PM_at       RNPC3    1 -5.572903 4.560983e-08 4.464407e-04
## 6    1566342_PM_at                6  5.559620 4.895630e-08 4.464407e-04
## 7     201705_PM_at       PSMD7   16  5.472279 7.771758e-08 4.694030e-04
## 8     239629_PM_at       CFLAR    2 -5.459875 8.294961e-08 4.694030e-04
## 9     235216_PM_at       ESCO1   18 -5.456719 8.433485e-08 4.694030e-04
## 10    244822_PM_at               21 -5.453456 8.579054e-08 4.694030e-04
## 11    241320_PM_at             &lt;NA&gt; -5.393997 1.170126e-07 5.820312e-04
## 12 1564190_PM_x_at      ZNF519   18 -5.226719 2.760750e-07 9.397599e-04
## 13  229501_PM_s_at        USP8   15  5.224467 2.792424e-07 9.397599e-04
## 14   1552348_PM_at      PRSS33   16  5.222501 2.820364e-07 9.397599e-04
## 15    219969_PM_at       TXLNG    X -5.219820 2.858894e-07 9.397599e-04
## 16    211074_PM_at       FOLR1   11 -5.209339 3.014495e-07 9.397599e-04
## 17    226219_PM_at    ARHGAP30    1  5.196301 3.219525e-07 9.397599e-04
## 18    229410_PM_at     SLC35E1   19  5.195339 3.235188e-07 9.397599e-04
## 19    202026_PM_at        SDHD   11  5.192892 3.275325e-07 9.397599e-04
## 20   1570352_PM_at         ATM   11 -5.171333 3.650549e-07 9.397599e-04
##      pos
## 1  18539
## 2  36536
## 3  41696
## 4  44895
## 5  36255
## 6   8251
## 7  11154
## 8  48879
## 9  44466
## 10 54073
## 11 50570
## 12  7509
## 13 38756
## 14    75
## 15 29254
## 16 20439
## 17 35476
## 18 38665
## 19 11475
## 20  9801
## 
## pvalue quantiles for intercept and covariates&gt;&gt;
##         (Intercept)   asthmaTRUE   severeTRUE    sexmale         gxPC1
## min    0.000000e+00 1.669291e-09 5.373778e-10 0.00000000 7.963565e-190
## 25%    8.007602e-84 8.711253e-02 1.253311e-01 0.06792468  1.237271e-28
## median 7.430805e-58 3.223858e-01 3.764067e-01 0.30368424  8.133367e-12
## 75%    9.385885e-41 6.412002e-01 6.824862e-01 0.63487767  5.389431e-04
## max    9.867523e-01 9.999485e-01 9.999636e-01 0.99998131  9.993542e-01
##               gxPC2      ESTNEUT
## min    3.807316e-90 4.130943e-67
## 25%    1.641704e-11 5.134678e-06
## median 7.226723e-05 1.216234e-02
## 75%    5.610325e-02 2.481584e-01
## max    9.999893e-01 9.997975e-01
## 
## formula&gt;&gt;
## ~asthma + severe + sex + gxPC1 + gxPC2 + ESTNEUT
## 
## covariate of interest is  asthma 
## Number of tests= 54715 
## Number of arrays= 410 
## Number of significant tests (raw p-value &lt;  0.05 )= 10377 
## Number of significant tests after p-value adjustments= 2479 
## 
## 
## **********************************************
## 
## &gt; # severe cases vs. moderate cases (no smokers)
## &gt; dge3 &lt;- lmFitWrapper(es_ubiopred_wb[, !es_ubiopred_wb$smoker &amp; es_ubiopred_wb$asthma], 
## +           .... [TRUNCATED] 
## dim(dat)&gt;&gt;
## [1] 54715   323
## 
## Running lmFit...
## Running eBayes...
## Preparing output...
##          probeIDs            geneSymbols chr     stats         pval
## 1    214370_PM_at                 S100A8   1  6.245505 1.335494e-09
## 2    201094_PM_at                  RPS29  14  6.211049 1.624991e-09
## 3    224972_PM_at                  ROMO1  20  6.182344 1.912365e-09
## 4  202917_PM_s_at                 S100A8   1  6.104139 2.971706e-09
## 5    218856_PM_at               TNFRSF21   6 -6.050945 4.001110e-09
## 6  217249_PM_x_at                          4  6.041401 4.219546e-09
## 7  200834_PM_s_at LOC100291837 /// RPS21  20  6.033136 4.418113e-09
## 8    201597_PM_at                 COX7A2   6  5.973954 6.132549e-09
## 9    210244_PM_at                   CAMP   3  5.961576 6.565868e-09
## 10 216262_PM_s_at                  TGIF2  20 -5.951425 6.943411e-09
## 11   226773_PM_at                          4 -5.901115 9.150742e-09
## 12 201492_PM_s_at                  RPL41  12  5.826253 1.375377e-08
## 13 217773_PM_s_at                 NDUFA4   7  5.821030 1.414831e-08
## 14 224841_PM_x_at                   GAS5   1  5.775051 1.813293e-08
## 15 214334_PM_x_at                 DAZAP2  12 -5.749909 2.075503e-08
## 16 207573_PM_x_at                  ATP5L  11  5.724923 2.372564e-08
## 17 224741_PM_x_at                   GAS5   1  5.707949 2.597618e-08
## 18 229465_PM_s_at                         19 -5.699246 2.720954e-08
## 19 226845_PM_s_at                 MYEOV2   2  5.673586 3.118739e-08
## 20   221474_PM_at                 MYL12B  18  5.650673 3.521448e-08
##           p.adj   pos
## 1  3.453387e-05 23670
## 2  3.453387e-05 10543
## 3  3.453387e-05 34231
## 4  3.453387e-05 12367
## 5  3.453387e-05 28141
## 6  3.453387e-05 26535
## 7  3.453387e-05 10283
## 8  3.799088e-05 11046
## 9  3.799088e-05 19645
## 10 3.799088e-05 25555
## 11 4.551662e-05 36029
## 12 5.954806e-05 10941
## 13 5.954806e-05 27059
## 14 7.086739e-05 34100
## 15 7.570744e-05 23634
## 16 8.113429e-05 17015
## 17 8.270944e-05 34001
## 18 8.270944e-05 38720
## 19 8.981149e-05 36101
## 20 9.633801e-05 30758
## 
## pvalue quantiles for intercept and covariates&gt;&gt;
##         (Intercept)   severeTRUE    sexmale         gxPC1        gxPC2
## min    0.000000e+00 1.335494e-09 0.00000000 3.521421e-149 8.929113e-70
## 25%    2.386617e-62 1.258925e-01 0.09499583  3.318426e-22 4.402135e-09
## median 1.825830e-42 3.809366e-01 0.34563960  3.140942e-09 4.550965e-04
## 75%    1.519464e-29 6.851606e-01 0.65991725  3.350454e-03 8.994666e-02
## max    9.856730e-01 9.999947e-01 0.99993437  9.999073e-01 9.999171e-01
##             ESTNEUT
## min    2.212534e-49
## 25%    4.328954e-05
## median 2.203923e-02
## 75%    2.904202e-01
## max    9.999777e-01
## 
## formula&gt;&gt;
## ~severe + sex + gxPC1 + gxPC2 + ESTNEUT
## 
## covariate of interest is  severe 
## Number of tests= 54715 
## Number of arrays= 323 
## Number of significant tests (raw p-value &lt;  0.05 )= 8178 
## Number of significant tests after p-value adjustments= 1447 
## 
## 
## **********************************************
## 
## &gt; ## ---- extract_dge_results
## &gt; dge1_rnks &lt;- dge1 %&gt;% use_series(frame) %&gt;% select(geneSymbols, stats) %&gt;% arrange(stats)
## 
## &gt; dge1_rnks &lt;- setNames(dge1_rnks$stats, dge1_rnks$geneSymbols)
## 
## &gt; dge2_rnks &lt;- dge2 %&gt;% use_series(frame) %&gt;% select(geneSymbols, stats) %&gt;% arrange(stats)
## 
## &gt; dge2_rnks &lt;- setNames(dge2_rnks$stats, dge2_rnks$geneSymbols)
## 
## &gt; dge3_rnks &lt;- dge3 %&gt;% use_series(frame) %&gt;% select(geneSymbols, stats) %&gt;% arrange(stats)
## 
## &gt; dge3_rnks &lt;- setNames(dge3_rnks$stats, dge3_rnks$geneSymbols)
## 
## &gt; ## ---- load_input_gene_sets
## &gt; # Focus on TREM1-related leading edge gene sets of asthma control
## &gt; # See Supp. Table 5 from Croteau-Chonka et al. (2 .... [TRUNCATED] 
## 
## &gt; gsc_c7_control_trem1 &lt;- gsc_c7_control[grepl(&quot;^GSE9988&quot;, names(gsc_c7_control))]
## 
## &gt; ## ---- run_gsea
## &gt; fgseaRes1 &lt;- fgsea(
## +     pathways = gsc_c7_control_trem1,
## +     stats = dge1_rnks,
## +     minSize = 15,
## +     maxSize = 500,
## +    .... [TRUNCATED] 
## 
## &gt; fgseaRes2 &lt;- fgsea(
## +     pathways = gsc_c7_control_trem1,
## +     stats = dge2_rnks,
## +     minSize = 15,
## +     maxSize = 500,
## +     nperm = 100000
## +  .... [TRUNCATED] 
## 
## &gt; fgseaRes3 &lt;- fgsea(
## +     pathways = gsc_c7_control_trem1,
## +     stats = dge3_rnks,
## +     minSize = 15,
## +     maxSize = 500,
## +     nperm = 100000
## +  .... [TRUNCATED]
</code></pre>

<pre><code class="r">trem1_pathways &lt;- fgseaRes1 %&gt;% arrange(desc(NES)) %&gt;% filter(padj &lt; 0.05) %&gt;% use_series(pathway)

pdf(file = here(&quot;reports&quot;, &quot;publication&quot;, &quot;Figure01.pdf&quot;), width = 8, height = 7, pointsize = 8)
plotGseaTable(pathways = gsc_c7_control[trem1_pathways],
              stats = dge1_rnks,
              fgseaRes = fgseaRes1,
              colwidths = c(8.0, 3.0, 0.8, 1.0, 1.0),
              gseaParam = 1)
dev.off()
</code></pre>

<pre><code>## png 
##   2
</code></pre>

<pre><code class="r"># How many cases and controls?
es_ubiopred_wb %&gt;% pData %$% table(asthma)
</code></pre>

<pre><code>## asthma
## FALSE  TRUE 
##    87   411
</code></pre>

<pre><code class="r">es_ubiopred_wb %&gt;% pData %$% table(severe, asthma)
</code></pre>

<pre><code>##        asthma
## severe  FALSE TRUE
##   FALSE    87   77
##   TRUE      0  334
</code></pre>

<pre><code class="r">es_ubiopred_wb %&gt;% pData %$% table(smoker, asthma)
</code></pre>

<pre><code>##        asthma
## smoker  FALSE TRUE
##   FALSE    87  323
##   TRUE      0   88
</code></pre>

<pre><code class="r">es_ubiopred_wb %&gt;% pData %&gt;% filter(asthma == TRUE) %$% table(severe)
</code></pre>

<pre><code>## severe
## FALSE  TRUE 
##    77   334
</code></pre>

<pre><code class="r">es_ubiopred_wb %&gt;% pData %&gt;% filter(asthma == TRUE) %$% table(smoker, severe)
</code></pre>

<pre><code>##        severe
## smoker  FALSE TRUE
##   FALSE    77  246
##   TRUE      0   88
</code></pre>

<pre><code class="r"># How many genes tested for differential expression?
dge1 %&gt;% use_series(frame) %&gt;% nrow
</code></pre>

<pre><code>## [1] 54715
</code></pre>

<pre><code class="r"># How many differentially expressed genes in each comparison?
dge1 %&gt;% use_series(frame) %&gt;% filter(p.adj &lt; 0.05) %&gt;% nrow # cases vs. ctrls. (all subjects)
</code></pre>

<pre><code>## [1] 2305
</code></pre>

<pre><code class="r">dge2 %&gt;% use_series(frame) %&gt;% filter(p.adj &lt; 0.05) %&gt;% nrow # cases vs. ctrls. (no smokers)
</code></pre>

<pre><code>## [1] 2479
</code></pre>

<pre><code class="r">dge3 %&gt;% use_series(frame) %&gt;% filter(p.adj &lt; 0.05) %&gt;% nrow # severe cases vs. moderate cases (no smokers)
</code></pre>

<pre><code>## [1] 1447
</code></pre>

<pre><code class="r"># How many TREM1 gene sets were tested?
fgseaRes1 %&gt;% use_series(pathway) %&gt;% length
</code></pre>

<pre><code>## [1] 24
</code></pre>

<pre><code class="r"># How large were the TREM-1 gene sets tested?
gsc_c7_control_trem1 %&gt;% sapply(length) %&gt;% summary
</code></pre>

<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   10.00   42.00   44.00   45.92   50.00   62.00
</code></pre>

<pre><code class="r"># How many TREM1 gene sets were enriched?
trem1_pathways %&gt;% length
</code></pre>

<pre><code>## [1] 24
</code></pre>

<pre><code class="r"># What are the core enriched TREM1 pathway genes?
core_genes &lt;- fgseaRes1 %&gt;% arrange(NES) %&gt;% filter(padj &lt; 0.05) %&gt;% use_series(leadingEdge) %&gt;% unlist %&gt;% unique %&gt;% sort
core_genes %&gt;% print
</code></pre>

<pre><code>##   [1] &quot;ACOX1&quot;        &quot;ACSL1&quot;        &quot;ACSL3&quot;        &quot;ADO&quot;         
##   [5] &quot;ADRB2&quot;        &quot;AKIRIN2&quot;      &quot;ANKRD57&quot;      &quot;AQP9&quot;        
##   [9] &quot;ARAP3&quot;        &quot;ARID5A&quot;       &quot;ARL5B&quot;        &quot;ARRB1&quot;       
##  [13] &quot;ATF3&quot;         &quot;BCAR3&quot;        &quot;BID&quot;          &quot;C10orf128&quot;   
##  [17] &quot;C11orf75&quot;     &quot;C14orf102&quot;    &quot;C17orf79&quot;     &quot;C1orf122&quot;    
##  [21] &quot;C1orf38&quot;      &quot;C6orf1&quot;       &quot;CALHM2&quot;       &quot;CASP5&quot;       
##  [25] &quot;CBL&quot;          &quot;CCL23&quot;        &quot;CCNG2&quot;        &quot;CCR2&quot;        
##  [29] &quot;CD300LB&quot;      &quot;CD68&quot;         &quot;CDKN1A&quot;       &quot;CEP350&quot;      
##  [33] &quot;CHST15&quot;       &quot;CLCF1&quot;        &quot;CLIC4&quot;        &quot;CMTM3&quot;       
##  [37] &quot;CRBN&quot;         &quot;CSNK1D&quot;       &quot;CTSB&quot;         &quot;CYBB&quot;        
##  [41] &quot;CYTH4&quot;        &quot;DAB2&quot;         &quot;DCP2&quot;         &quot;DDX17&quot;       
##  [45] &quot;DOK2&quot;         &quot;DR1&quot;          &quot;DRAM1&quot;        &quot;DUSP18&quot;      
##  [49] &quot;DUSP3&quot;        &quot;DYRK3&quot;        &quot;EFR3A&quot;        &quot;EHD1&quot;        
##  [53] &quot;ELF2&quot;         &quot;ENG&quot;          &quot;EPB41L3&quot;      &quot;ETS2&quot;        
##  [57] &quot;EVI2B&quot;        &quot;FAM105A&quot;      &quot;FKBP15&quot;       &quot;FOXN2&quot;       
##  [61] &quot;FRAT1&quot;        &quot;FUCA1&quot;        &quot;GAPT&quot;         &quot;GBP1&quot;        
##  [65] &quot;GDI2&quot;         &quot;GFOD1&quot;        &quot;GK&quot;           &quot;GLA&quot;         
##  [69] &quot;GRB2&quot;         &quot;GRN&quot;          &quot;HBEGF&quot;        &quot;HCK&quot;         
##  [73] &quot;HECA&quot;         &quot;HHEX&quot;         &quot;HSD3B7&quot;       &quot;IER5&quot;        
##  [77] &quot;IFI30&quot;        &quot;IFIH1&quot;        &quot;IFNGR2&quot;       &quot;IL18&quot;        
##  [81] &quot;IL1B&quot;         &quot;ITGAX&quot;        &quot;JMJD1C&quot;       &quot;KCNJ2&quot;       
##  [85] &quot;KLF6&quot;         &quot;KMO&quot;          &quot;LAIR1&quot;        &quot;LDLR&quot;        
##  [89] &quot;LFNG&quot;         &quot;LOC100129034&quot; &quot;LRRC25&quot;       &quot;LRRC33&quot;      
##  [93] &quot;LYSMD2&quot;       &quot;MAP3K14&quot;      &quot;MAP3K8&quot;       &quot;MAPK13&quot;      
##  [97] &quot;MAPKAPK3&quot;     &quot;MARCKS&quot;       &quot;MBD2&quot;         &quot;MKL1&quot;        
## [101] &quot;MR1&quot;          &quot;MXD1&quot;         &quot;MYD88&quot;        &quot;N4BP1&quot;       
## [105] &quot;NACC2&quot;        &quot;NCKAP1L&quot;      &quot;NCOR2&quot;        &quot;NFATC1&quot;      
## [109] &quot;NFIC&quot;         &quot;NFKBIZ&quot;       &quot;NIN&quot;          &quot;NINJ1&quot;       
## [113] &quot;NPEPPS&quot;       &quot;OASL&quot;         &quot;OLIG1&quot;        &quot;OLIG2&quot;       
## [117] &quot;PARP10&quot;       &quot;PDK4&quot;         &quot;PELI1&quot;        &quot;PHACTR2&quot;     
## [121] &quot;PIK3AP1&quot;      &quot;PILRA&quot;        &quot;PLAUR&quot;        &quot;PLEK&quot;        
## [125] &quot;PLSCR1&quot;       &quot;PLXNB2&quot;       &quot;PMAIP1&quot;       &quot;PPP2R5A&quot;     
## [129] &quot;PSTPIP2&quot;      &quot;PTAFR&quot;        &quot;PTGER4&quot;       &quot;PTGS2&quot;       
## [133] &quot;PTPN6&quot;        &quot;RAB11FIP4&quot;    &quot;RAB31&quot;        &quot;RAPGEF2&quot;     
## [137] &quot;RARA&quot;         &quot;RASSF2&quot;       &quot;RCOR3&quot;        &quot;RGS1&quot;        
## [141] &quot;RGS12&quot;        &quot;RGS19&quot;        &quot;RHOB&quot;         &quot;RHOBTB3&quot;     
## [145] &quot;RHOH&quot;         &quot;RHOU&quot;         &quot;RIN2&quot;         &quot;RIN3&quot;        
## [149] &quot;RIT1&quot;         &quot;RNF144B&quot;      &quot;RNF19B&quot;       &quot;RRAGC&quot;       
## [153] &quot;RRAGD&quot;        &quot;RYBP&quot;         &quot;S1PR3&quot;        &quot;SCARB2&quot;      
## [157] &quot;SCO2&quot;         &quot;SERPINB1&quot;     &quot;SERPINB2&quot;     &quot;SESTD1&quot;      
## [161] &quot;SH2D3C&quot;       &quot;SIRPA&quot;        &quot;SLA&quot;          &quot;SLC16A3&quot;     
## [165] &quot;SLC2A3&quot;       &quot;SLC2A6&quot;       &quot;SLC38A2&quot;      &quot;SLC43A2&quot;     
## [169] &quot;SNX20&quot;        &quot;SOD2&quot;         &quot;SORL1&quot;        &quot;SPAG9&quot;       
## [173] &quot;ST3GAL6&quot;      &quot;ST8SIA4&quot;      &quot;STK17B&quot;       &quot;STX11&quot;       
## [177] &quot;STX3&quot;         &quot;SYN2&quot;         &quot;TAF5&quot;         &quot;TGFBR1&quot;      
## [181] &quot;THBS1&quot;        &quot;TIFA&quot;         &quot;TIMP2&quot;        &quot;TLR1&quot;        
## [185] &quot;TLR4&quot;         &quot;TLR7&quot;         &quot;TMBIM1&quot;       &quot;TMEM123&quot;     
## [189] &quot;TMEM88&quot;       &quot;TNF&quot;          &quot;TNFAIP2&quot;      &quot;TNFSF13B&quot;    
## [193] &quot;TNRC6A&quot;       &quot;TOR1B&quot;        &quot;TRIB1&quot;        &quot;TRIM33&quot;      
## [197] &quot;TRIM38&quot;       &quot;TSC22D3&quot;      &quot;UBE2W&quot;        &quot;USP22&quot;       
## [201] &quot;VAV1&quot;         &quot;WDFY1&quot;        &quot;ZBTB2&quot;        &quot;ZC3HAV1&quot;     
## [205] &quot;ZFP36&quot;        &quot;ZFYVE16&quot;      &quot;ZNF227&quot;       &quot;ZNF398&quot;
</code></pre>

<pre><code class="r">core_genes %&gt;% length
</code></pre>

<pre><code>## [1] 208
</code></pre>

<pre><code class="r"># What is the overlap of this core with highlighted TREM1 genes from Table 3 of Croteau-Chonka et al. (2017)?
table3_genes &lt;- c(&quot;CCL23&quot;, &quot;OLIG1&quot;, &quot;OLIG2&quot;, &quot;GFOD1&quot;, &quot;RHOBTB3&quot;, &quot;HSD3B7&quot;)
intersect(core_genes, table3_genes)
</code></pre>

<pre><code>## [1] &quot;CCL23&quot;   &quot;GFOD1&quot;   &quot;HSD3B7&quot;  &quot;OLIG1&quot;   &quot;OLIG2&quot;   &quot;RHOBTB3&quot;
</code></pre>

<pre><code class="r">dge1 %&gt;% use_series(frame) %&gt;% filter(geneSymbols %in% table3_genes)
</code></pre>

<pre><code>##          probeIDs geneSymbols chr      stats         pval       p.adj
## 1  210549_PM_s_at       CCL23  17  4.2643240 2.402434e-05 0.005134733
## 2    213825_PM_at       OLIG2  21  4.1486384 3.936892e-05 0.006483616
## 3    210548_PM_at       CCL23  17  4.1070741 4.688221e-05 0.007001643
## 4    228170_PM_at       OLIG1  21  2.7698083 5.819737e-03 0.085571990
## 5    222817_PM_at      HSD3B7  16  2.7328801 6.503029e-03 0.090838205
## 6  219821_PM_s_at       GFOD1   6  2.1847211 2.937779e-02 0.204857411
## 7  202976_PM_s_at     RHOBTB3   5  2.1198299 3.451761e-02 0.222545129
## 8    225202_PM_at     RHOBTB3   5  1.9700836 4.938602e-02 0.268121678
## 9    216049_PM_at     RHOBTB3   5 -1.5757498 1.157226e-01 0.407934595
## 10   213824_PM_at       OLIG2  21  1.3853653 1.665650e-01 0.483765151
## 11 216048_PM_s_at     RHOBTB3   5  0.3561856 7.218534e-01 0.895618405
## 12   240111_PM_at     RHOBTB3   5 -0.2917926 7.705675e-01 0.915899020
## 13 202975_PM_s_at     RHOBTB3   5 -0.1752128 8.609841e-01 0.951746043
##      pos
## 1  19938
## 2  23126
## 3  19937
## 4  37425
## 5  32097
## 6  29106
## 7  12426
## 8  34460
## 9  25342
## 10 23125
## 11 25341
## 12 49361
## 13 12425
</code></pre>

<pre><code class="r"># Which primary analysis gene sets were enriched in secondary analyses?
fgseaRes1 %&gt;% arrange(desc(NES)) %&gt;% select(-leadingEdge) %&gt;% filter(padj &lt; 0.05)
</code></pre>

<pre><code>##                                                       pathway         pval
## 1               GSE9988_LPS_VS_LPS_AND_ANTI_TREM1_MONOCYTE_UP 2.143577e-05
## 2     GSE9988_ANTI_TREM1_AND_LPS_VS_CTRL_TREATED_MONOCYTES_DN 2.127931e-05
## 3             GSE9988_ANTI_TREM1_VS_CTRL_TREATED_MONOCYTES_DN 2.133333e-05
## 4                 GSE9988_LOW_LPS_VS_CTRL_TREATED_MONOCYTE_DN 2.133333e-05
## 5           GSE9988_LOW_LPS_VS_ANTI_TREM1_AND_LPS_MONOCYTE_UP 2.145278e-05
## 6              GSE9988_LOW_LPS_VS_VEHICLE_TREATED_MONOCYTE_DN 2.134381e-05
## 7  GSE9988_ANTI_TREM1_AND_LPS_VS_VEHICLE_TREATED_MONOCYTES_DN 2.133333e-05
## 8                   GSE9988_ANTI_TREM1_VS_LOW_LPS_MONOCYTE_UP 2.138397e-05
## 9                       GSE9988_ANTI_TREM1_VS_LPS_MONOCYTE_UP 2.142704e-05
## 10                  GSE9988_ANTI_TREM1_VS_LOW_LPS_MONOCYTE_DN 2.154708e-05
## 11         GSE9988_ANTI_TREM1_VS_VEHICLE_TREATED_MONOCYTES_DN 2.123999e-05
## 12                      GSE9988_ANTI_TREM1_VS_LPS_MONOCYTE_DN 2.154708e-05
## 13                 GSE9988_LPS_VS_VEHICLE_TREATED_MONOCYTE_DN 2.135201e-05
## 14          GSE9988_LOW_LPS_VS_ANTI_TREM1_AND_LPS_MONOCYTE_DN 2.135201e-05
## 15              GSE9988_LPS_VS_LPS_AND_ANTI_TREM1_MONOCYTE_DN 2.133333e-05
## 16       GSE9988_ANTI_TREM1_VS_ANTI_TREM1_AND_LPS_MONOCYTE_DN 2.144082e-05
## 17            GSE9988_ANTI_TREM1_VS_CTRL_TREATED_MONOCYTES_UP 6.400000e-05
## 18                    GSE9988_LPS_VS_CTRL_TREATED_MONOCYTE_UP 1.495886e-04
## 19    GSE9988_ANTI_TREM1_AND_LPS_VS_CTRL_TREATED_MONOCYTES_UP 3.632634e-04
## 20                 GSE9988_LPS_VS_VEHICLE_TREATED_MONOCYTE_UP 8.115496e-04
## 21             GSE9988_LOW_LPS_VS_VEHICLE_TREATED_MONOCYTE_UP 1.067395e-03
## 22                GSE9988_LOW_LPS_VS_CTRL_TREATED_MONOCYTE_UP 1.856000e-03
## 23         GSE9988_ANTI_TREM1_VS_VEHICLE_TREATED_MONOCYTES_UP 3.822010e-03
## 24 GSE9988_ANTI_TREM1_AND_LPS_VS_VEHICLE_TREATED_MONOCYTES_UP 5.796424e-03
##            padj        ES      NES nMoreExtreme size
## 1  3.232062e-05 0.7462695 3.037452            0   59
## 2  3.232062e-05 0.7706867 2.892690            0   41
## 3  3.232062e-05 0.7525535 2.869805            0   44
## 4  3.232062e-05 0.7380543 2.814513            0   44
## 5  3.232062e-05 0.6914557 2.771448            0   55
## 6  3.232062e-05 0.7025062 2.664984            0   43
## 7  3.232062e-05 0.6843808 2.609833            0   44
## 8  3.232062e-05 0.6660411 2.590442            0   48
## 9  3.232062e-05 0.6458536 2.557266            0   52
## 10 3.232062e-05 0.6216615 2.553596            0   62
## 11 3.232062e-05 0.6833943 2.551336            0   40
## 12 3.232062e-05 0.6160032 2.530353            0   62
## 13 3.232062e-05 0.6679562 2.519729            0   42
## 14 3.232062e-05 0.6294299 2.374397            0   42
## 15 3.232062e-05 0.6052969 2.308253            0   44
## 16 3.232062e-05 0.5334658 2.162636            0   58
## 17 9.035294e-05 0.5517037 2.103880            2   44
## 18 1.994515e-04 0.5247718 2.060198            6   50
## 19 4.588590e-04 0.4925250 1.925896           16   49
## 20 9.738596e-04 0.4833068 1.861876           37   46
## 21 1.219880e-03 0.4796964 1.839242           49   45
## 22 2.024727e-03 0.4715886 1.798367           86   44
## 23 3.988184e-03 0.4599951 1.735238          178   42
## 24 5.796424e-03 0.4582767 1.691799          272   38
</code></pre>

<pre><code class="r">fgseaRes2 %&gt;% filter(pathway %in% trem1_pathways) %&gt;% select(-leadingEdge) %&gt;% filter(padj &lt; 0.05)
</code></pre>

<pre><code>##                                                       pathway         pval
## 1     GSE9988_ANTI_TREM1_AND_LPS_VS_CTRL_TREATED_MONOCYTES_DN 2.136250e-05
## 2     GSE9988_ANTI_TREM1_AND_LPS_VS_CTRL_TREATED_MONOCYTES_UP 2.361579e-04
## 3  GSE9988_ANTI_TREM1_AND_LPS_VS_VEHICLE_TREATED_MONOCYTES_DN 2.134472e-05
## 4  GSE9988_ANTI_TREM1_AND_LPS_VS_VEHICLE_TREATED_MONOCYTES_UP 2.960218e-03
## 5        GSE9988_ANTI_TREM1_VS_ANTI_TREM1_AND_LPS_MONOCYTE_DN 2.159081e-05
## 6             GSE9988_ANTI_TREM1_VS_CTRL_TREATED_MONOCYTES_DN 2.134472e-05
## 7             GSE9988_ANTI_TREM1_VS_CTRL_TREATED_MONOCYTES_UP 8.537887e-05
## 8                   GSE9988_ANTI_TREM1_VS_LOW_LPS_MONOCYTE_DN 2.164783e-05
## 9                   GSE9988_ANTI_TREM1_VS_LOW_LPS_MONOCYTE_UP 2.138626e-05
## 10                      GSE9988_ANTI_TREM1_VS_LPS_MONOCYTE_DN 2.164783e-05
## 11                      GSE9988_ANTI_TREM1_VS_LPS_MONOCYTE_UP 2.151556e-05
## 12         GSE9988_ANTI_TREM1_VS_VEHICLE_TREATED_MONOCYTES_DN 2.131469e-05
## 13         GSE9988_ANTI_TREM1_VS_VEHICLE_TREATED_MONOCYTES_UP 2.408817e-03
## 14          GSE9988_LOW_LPS_VS_ANTI_TREM1_AND_LPS_MONOCYTE_DN 2.131696e-05
## 15          GSE9988_LOW_LPS_VS_ANTI_TREM1_AND_LPS_MONOCYTE_UP 2.156892e-05
## 16                GSE9988_LOW_LPS_VS_CTRL_TREATED_MONOCYTE_DN 2.134472e-05
## 17                GSE9988_LOW_LPS_VS_CTRL_TREATED_MONOCYTE_UP 1.664888e-03
## 18             GSE9988_LOW_LPS_VS_VEHICLE_TREATED_MONOCYTE_DN 2.130379e-05
## 19             GSE9988_LOW_LPS_VS_VEHICLE_TREATED_MONOCYTE_UP 8.330663e-04
## 20                    GSE9988_LPS_VS_CTRL_TREATED_MONOCYTE_UP 4.287245e-05
## 21              GSE9988_LPS_VS_LPS_AND_ANTI_TREM1_MONOCYTE_DN 2.134472e-05
## 22              GSE9988_LPS_VS_LPS_AND_ANTI_TREM1_MONOCYTE_UP 2.164081e-05
## 23                 GSE9988_LPS_VS_VEHICLE_TREATED_MONOCYTE_DN 2.131696e-05
## 24                 GSE9988_LPS_VS_VEHICLE_TREATED_MONOCYTE_UP 4.277617e-04
##            padj        ES      NES nMoreExtreme size
## 1  3.247175e-05 0.7779490 2.925557            0   41
## 2  2.983047e-04 0.5003731 1.957904           10   49
## 3  3.247175e-05 0.6895618 2.635430            0   44
## 4  2.960218e-03 0.4743669 1.752870          138   38
## 5  3.247175e-05 0.5455014 2.211442            0   58
## 6  3.247175e-05 0.7461772 2.851808            0   44
## 7  1.138385e-04 0.5545600 2.119468            3   44
## 8  3.247175e-05 0.6244654 2.566773            0   62
## 9  3.247175e-05 0.6680436 2.603946            0   48
## 10 3.247175e-05 0.6184207 2.541927            0   62
## 11 3.247175e-05 0.6462016 2.559111            0   52
## 12 3.247175e-05 0.6911018 2.584515            0   40
## 13 2.513548e-03 0.4693161 1.775373          112   42
## 14 3.247175e-05 0.6235836 2.358951            0   42
## 15 3.247175e-05 0.6955706 2.788095            0   55
## 16 3.247175e-05 0.7472163 2.855779            0   44
## 17 1.816241e-03 0.4748922 1.814986           77   44
## 18 3.247175e-05 0.7127222 2.710032            0   43
## 19 9.520758e-04 0.4854217 1.864894           38   45
## 20 6.052582e-05 0.5356944 2.105468            1   50
## 21 3.247175e-05 0.6012046 2.297739            0   44
## 22 3.247175e-05 0.7475125 3.039678            0   59
## 23 3.247175e-05 0.6841896 2.588217            0   42
## 24 5.133141e-04 0.4960656 1.914845           19   46
</code></pre>

<pre><code class="r">fgseaRes3 %&gt;% filter(pathway %in% trem1_pathways) %&gt;% select(-leadingEdge) %&gt;% filter(padj &lt; 0.05)
</code></pre>

<pre><code>##                                                       pathway         pval
## 1     GSE9988_ANTI_TREM1_AND_LPS_VS_CTRL_TREATED_MONOCYTES_UP 1.290123e-04
## 2  GSE9988_ANTI_TREM1_AND_LPS_VS_VEHICLE_TREATED_MONOCYTES_UP 2.508466e-05
## 3             GSE9988_ANTI_TREM1_VS_CTRL_TREATED_MONOCYTES_DN 5.755616e-03
## 4             GSE9988_ANTI_TREM1_VS_CTRL_TREATED_MONOCYTES_UP 1.375236e-03
## 5                   GSE9988_ANTI_TREM1_VS_LOW_LPS_MONOCYTE_UP 2.405578e-02
## 6                       GSE9988_ANTI_TREM1_VS_LPS_MONOCYTE_UP 1.684282e-02
## 7          GSE9988_ANTI_TREM1_VS_VEHICLE_TREATED_MONOCYTES_UP 5.062522e-05
## 8           GSE9988_LOW_LPS_VS_ANTI_TREM1_AND_LPS_MONOCYTE_DN 1.769351e-02
## 9                 GSE9988_LOW_LPS_VS_CTRL_TREATED_MONOCYTE_UP 1.298834e-02
## 10             GSE9988_LOW_LPS_VS_VEHICLE_TREATED_MONOCYTE_UP 2.814307e-03
## 11                    GSE9988_LPS_VS_CTRL_TREATED_MONOCYTE_UP 8.170653e-03
## 12              GSE9988_LPS_VS_LPS_AND_ANTI_TREM1_MONOCYTE_DN 5.857485e-04
## 13                 GSE9988_LPS_VS_VEHICLE_TREATED_MONOCYTE_UP 4.254556e-03
##            padj        ES      NES nMoreExtreme size
## 1  0.0010320983 0.5160203 2.029216            4   49
## 2  0.0006020319 0.5956238 2.211400            0   38
## 3  0.0172668466 0.4357837 1.671790          225   44
## 4  0.0066011307 0.4735213 1.816563           53   44
## 5  0.0444106681 0.3813409 1.492764          934   48
## 6  0.0353870298 0.3835667 1.528109          646   52
## 7  0.0006075027 0.5565813 2.113514            1   42
## 8  0.0353870298 0.4083608 1.550674          698   42
## 9  0.0311720063 0.4104216 1.574494          509   44
## 10 0.0112572277 0.4505652 1.736675          109   45
## 11 0.0217884077 0.4098414 1.619508          315   50
## 12 0.0035144909 0.4990881 1.914644           22   44
## 13 0.0145870481 0.4385599 1.698821          165   46
</code></pre>

<pre><code class="r">session_info()
</code></pre>

<pre><code>## Session info -------------------------------------------------------------
</code></pre>

<pre><code>##  setting  value                       
##  version  R version 3.4.0 (2017-04-21)
##  system   x86_64, linux-gnu           
##  ui       RStudio (1.0.143)           
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  tz       &lt;NA&gt;                        
##  date     2017-09-26
</code></pre>

<pre><code>## Packages -----------------------------------------------------------------
</code></pre>

<pre><code>##  package              * version  date       source                      
##  affy                   1.54.0   2017-05-02 Bioconductor                
##  affyio                 1.46.0   2017-05-02 Bioconductor                
##  annotate             * 1.54.0   2017-05-02 Bioconductor                
##  AnnotationDbi        * 1.38.2   2017-08-16 Bioconductor                
##  assertthat             0.2.0    2017-04-11 CRAN (R 3.4.0)              
##  backports              1.1.0    2017-05-22 CRAN (R 3.4.0)              
##  base                 * 3.4.0    2017-05-02 local                       
##  base64                 2.0      2016-05-10 CRAN (R 3.4.0)              
##  beanplot               1.2      2014-09-19 CRAN (R 3.4.0)              
##  beeswarm               0.2.3    2016-04-25 CRAN (R 3.4.0)              
##  bibtex                 0.4.2    2017-06-30 CRAN (R 3.4.0)              
##  bigmemory            * 4.5.19   2016-03-28 CRAN (R 3.4.0)              
##  bigmemory.sri        * 0.1.3    2014-08-18 CRAN (R 3.4.0)              
##  bindr                  0.1      2016-11-13 CRAN (R 3.4.0)              
##  bindrcpp             * 0.2      2017-06-17 CRAN (R 3.4.0)              
##  Biobase              * 2.36.2   2017-07-19 Bioconductor                
##  BiocGenerics         * 0.22.0   2017-05-02 Bioconductor                
##  BiocInstaller          1.26.0   2017-05-02 Bioconductor                
##  BiocParallel           1.10.1   2017-07-19 Bioconductor                
##  biomaRt                2.32.1   2017-07-19 Bioconductor                
##  Biostrings             2.44.2   2017-07-25 Bioconductor                
##  bit                    1.1-12   2014-04-09 CRAN (R 3.4.0)              
##  bit64                  0.9-7    2017-05-08 CRAN (R 3.4.0)              
##  bitops                 1.0-6    2013-08-17 CRAN (R 3.4.0)              
##  blob                   1.1.0    2017-06-17 CRAN (R 3.4.0)              
##  broom                * 0.4.2    2017-02-13 CRAN (R 3.4.0)              
##  bumphunter             1.16.0   2017-05-02 Bioconductor                
##  caTools                1.17.1   2014-09-10 CRAN (R 3.4.0)              
##  CellMix              * 1.6.2    2017-07-25 local                       
##  cluster              * 2.0.6    2017-03-16 CRAN (R 3.4.0)              
##  codetools              0.2-15   2016-10-05 CRAN (R 3.4.0)              
##  colorspace             1.3-2    2016-12-14 CRAN (R 3.4.0)              
##  compiler             * 3.4.0    2017-05-02 local                       
##  corpcor                1.6.9    2017-04-01 CRAN (R 3.4.0)              
##  csSAM                * 1.2.4    2013-05-13 CRAN (R 3.4.0)              
##  data.table             1.10.4   2017-02-01 CRAN (R 3.4.0)              
##  datasets             * 3.4.0    2017-05-02 local                       
##  DBI                    0.7      2017-06-18 CRAN (R 3.4.0)              
##  DelayedArray           0.2.7    2017-07-19 Bioconductor                
##  devtools             * 1.13.3   2017-08-02 CRAN (R 3.4.0)              
##  digest                 0.6.12   2017-01-27 CRAN (R 3.4.0)              
##  doParallel             1.0.10   2015-10-14 CRAN (R 3.4.0)              
##  doRNG                  1.6.6    2017-04-10 CRAN (R 3.4.0)              
##  dplyr                * 0.7.2    2017-07-20 CRAN (R 3.4.0)              
##  evaluate               0.10.1   2017-06-24 CRAN (R 3.4.0)              
##  ezknitr              * 0.6      2016-09-16 CRAN (R 3.4.0)              
##  fastmatch              1.1-0    2017-01-28 CRAN (R 3.4.0)              
##  fgsea                * 1.2.1    2017-07-18 Bioconductor                
##  forcats              * 0.2.0    2017-01-23 CRAN (R 3.4.0)              
##  foreach                1.4.3    2015-10-13 CRAN (R 3.4.0)              
##  foreign                0.8-69   2017-06-21 CRAN (R 3.4.0)              
##  gdata                  2.18.0   2017-06-06 CRAN (R 3.4.0)              
##  genefilter           * 1.58.1   2017-07-19 Bioconductor                
##  GeneSelectMMD          2.20.1   2017-07-07 Bioconductor                
##  GenomeInfoDb           1.12.2   2017-07-19 Bioconductor                
##  GenomeInfoDbData       0.99.0   2017-05-02 Bioconductor                
##  GenomicAlignments      1.12.1   2017-07-19 Bioconductor                
##  GenomicFeatures        1.28.4   2017-07-19 Bioconductor                
##  GenomicRanges          1.28.4   2017-07-19 Bioconductor                
##  GEOquery               2.42.0   2017-05-02 Bioconductor                
##  ggplot2                2.2.1    2016-12-30 CRAN (R 3.4.0)              
##  glue                   1.1.1    2017-06-21 CRAN (R 3.4.0)              
##  gplots               * 3.0.1    2016-03-30 CRAN (R 3.4.0)              
##  graph                * 1.54.0   2017-05-02 Bioconductor                
##  graphics             * 3.4.0    2017-05-02 local                       
##  grDevices            * 3.4.0    2017-05-02 local                       
##  grid                   3.4.0    2017-05-02 local                       
##  gridBase               0.4-7    2014-02-24 CRAN (R 3.4.0)              
##  gridExtra              2.2.1    2016-02-29 CRAN (R 3.4.0)              
##  GSEABase             * 1.38.0   2017-07-18 Bioconductor                
##  gtable                 0.2.0    2016-02-26 CRAN (R 3.4.0)              
##  gtools                 3.5.0    2015-05-29 CRAN (R 3.4.0)              
##  here                 * 0.1      2017-09-26 Github (krlmlr/here@93593ee)
##  hgu133a.db           * 3.2.3    2017-07-18 Bioconductor                
##  hgu133b.db           * 3.2.3    2017-07-25 Bioconductor                
##  hthgu133pluspm.db    * 0.1      2017-07-20 Bioconductor                
##  htmltools              0.3.6    2017-04-28 CRAN (R 3.4.0)              
##  htmlwidgets            0.9      2017-07-10 CRAN (R 3.4.0)              
##  httpuv                 1.3.5    2017-07-04 CRAN (R 3.4.0)              
##  httr                   1.3.0    2017-08-16 CRAN (R 3.4.0)              
##  iCheck               * 1.6.0    2017-07-18 Bioconductor                
##  illuminaio             0.18.0   2017-05-02 Bioconductor                
##  IRanges              * 2.10.2   2017-07-19 Bioconductor                
##  iterators              1.0.8    2015-10-13 CRAN (R 3.4.0)              
##  jsonlite               1.5      2017-06-01 CRAN (R 3.4.0)              
##  KernSmooth             2.23-15  2015-06-29 CRAN (R 3.4.0)              
##  knitr                  1.17     2017-08-10 CRAN (R 3.4.0)              
##  labeling               0.3      2014-08-23 CRAN (R 3.4.0)              
##  lattice                0.20-35  2017-03-25 CRAN (R 3.4.0)              
##  lazyeval               0.2.0    2016-06-12 CRAN (R 3.4.0)              
##  limma                  3.32.5   2017-08-16 Bioconductor                
##  limSolve               1.5.5.3  2017-08-14 CRAN (R 3.4.0)              
##  lmtest                 0.9-35   2017-02-11 CRAN (R 3.4.0)              
##  locfit                 1.5-9.1  2013-04-20 CRAN (R 3.4.0)              
##  lpSolve                5.6.13   2015-09-19 CRAN (R 3.4.0)              
##  lumi                 * 2.28.0   2017-05-02 Bioconductor                
##  magrittr             * 1.5      2014-11-22 CRAN (R 3.4.0)              
##  MASS                   7.3-47   2017-04-21 CRAN (R 3.4.0)              
##  Matrix                 1.2-11   2017-08-16 CRAN (R 3.4.0)              
##  matrixStats            0.52.2   2017-04-14 CRAN (R 3.4.0)              
##  mclust                 5.3      2017-05-21 CRAN (R 3.4.0)              
##  memoise                1.1.0    2017-04-21 CRAN (R 3.4.0)              
##  methods              * 3.4.0    2017-05-02 local                       
##  methylumi              2.22.0   2017-05-02 Bioconductor                
##  mgcv                   1.8-18   2017-07-28 CRAN (R 3.4.0)              
##  mime                   0.5      2016-07-07 CRAN (R 3.4.0)              
##  minfi                  1.22.1   2017-07-19 Bioconductor                
##  mnormt                 1.5-5    2016-10-15 CRAN (R 3.4.0)              
##  multtest               2.32.0   2017-05-03 Bioconductor                
##  munsell                0.4.3    2016-02-13 CRAN (R 3.4.0)              
##  nleqslv                3.3.1    2017-07-06 CRAN (R 3.4.0)              
##  nlme                   3.1-131  2017-02-06 CRAN (R 3.4.0)              
##  NMF                  * 0.20.6   2015-05-26 CRAN (R 3.4.0)              
##  nor1mix                1.2-2    2016-08-25 CRAN (R 3.4.0)              
##  openssl                0.9.6    2016-12-31 CRAN (R 3.4.0)              
##  org.Hs.eg.db         * 3.4.1    2017-05-03 Bioconductor                
##  parallel             * 3.4.0    2017-05-02 local                       
##  pkgconfig              2.0.1    2017-03-21 CRAN (R 3.4.0)              
##  pkgmaker             * 0.22     2014-05-14 CRAN (R 3.4.0)              
##  plyr                   1.8.4    2016-06-08 CRAN (R 3.4.0)              
##  preprocessCore         1.38.1   2017-07-19 Bioconductor                
##  psych                  1.7.5    2017-05-03 CRAN (R 3.4.0)              
##  purrr                  0.2.3    2017-08-02 CRAN (R 3.4.0)              
##  quadprog               1.5-5    2013-04-17 CRAN (R 3.4.0)              
##  R.methodsS3            1.7.1    2016-02-16 CRAN (R 3.4.0)              
##  R.oo                   1.21.0   2016-11-01 CRAN (R 3.4.0)              
##  R.utils                2.5.0    2016-11-07 CRAN (R 3.4.0)              
##  R6                     2.2.2    2017-06-17 CRAN (R 3.4.0)              
##  randomForest           4.6-12   2015-10-07 CRAN (R 3.4.0)              
##  RColorBrewer           1.1-2    2014-12-07 CRAN (R 3.4.0)              
##  Rcpp                 * 0.12.12  2017-07-15 CRAN (R 3.4.0)              
##  RCurl                  1.95-4.8 2016-03-01 CRAN (R 3.4.0)              
##  registry             * 0.3      2015-07-08 CRAN (R 3.4.0)              
##  reshape                0.8.7    2017-08-06 CRAN (R 3.4.0)              
##  reshape2               1.4.2    2016-10-22 CRAN (R 3.4.0)              
##  rgl                    0.98.1   2017-03-08 CRAN (R 3.4.0)              
##  rlang                  0.1.2    2017-08-09 CRAN (R 3.4.0)              
##  rngtools             * 1.2.4    2014-03-06 CRAN (R 3.4.0)              
##  rprojroot              1.2      2017-01-16 CRAN (R 3.4.0)              
##  Rsamtools              1.28.0   2017-05-03 Bioconductor                
##  RSQLite                2.0      2017-06-19 CRAN (R 3.4.0)              
##  rtracklayer            1.36.4   2017-07-19 Bioconductor                
##  S4Vectors            * 0.14.3   2017-07-19 Bioconductor                
##  scales                 0.4.1    2016-11-09 CRAN (R 3.4.0)              
##  scatterplot3d          0.3-40   2017-04-22 CRAN (R 3.4.0)              
##  shiny                  1.0.4    2017-08-14 CRAN (R 3.4.0)              
##  siggenes               1.50.0   2017-05-03 Bioconductor                
##  splines                3.4.0    2017-05-02 local                       
##  stats                * 3.4.0    2017-05-02 local                       
##  stats4               * 3.4.0    2017-05-02 local                       
##  stringi                1.1.5    2017-04-07 CRAN (R 3.4.0)              
##  stringr              * 1.2.0    2017-02-18 CRAN (R 3.4.0)              
##  SummarizedExperiment   1.6.3    2017-07-19 Bioconductor                
##  survival               2.41-3   2017-04-04 CRAN (R 3.4.0)              
##  tibble                 1.3.3    2017-05-28 CRAN (R 3.4.0)              
##  tidyr                  0.7.0    2017-08-16 CRAN (R 3.4.0)              
##  tools                  3.4.0    2017-05-02 local                       
##  utils                * 3.4.0    2017-05-02 local                       
##  withr                  2.0.0    2017-07-28 CRAN (R 3.4.0)              
##  XML                  * 3.98-1.9 2017-06-19 CRAN (R 3.4.0)              
##  xtable                 1.8-2    2016-02-05 CRAN (R 3.4.0)              
##  XVector                0.16.0   2017-05-03 Bioconductor                
##  zlibbioc               1.22.0   2017-05-03 Bioconductor                
##  zoo                    1.8-0    2017-04-12 CRAN (R 3.4.0)
</code></pre>

</body>

</html>
